File: ContenidoAlumniApi/Content/models/Evento.py
from django.contrib.postgres.fields import ArrayField
from django.db import models


class Evento(models.Model):
    class EstadoEvento(models.TextChoices):
        ACTIVO = 'ACTIVO', 'Activo'
        FINALIZADO = 'FINALIZADO', 'Finalizado'
        CANCELADO = 'CANCELADO', 'Cancelado'

    titulo = models.CharField(max_length=255)
    descripcion = models.TextField(blank=True)

    fecha_inicio = models.DateTimeField()
    fecha_fin = models.DateTimeField(null=True, blank=True)

    carreras = ArrayField(models.IntegerField(), blank=True, default=list)
    usuarios = ArrayField(models.IntegerField(), blank=True, default=list)

    requiere_registro = models.BooleanField(default=False)
    estado = models.CharField(max_length=20, choices=EstadoEvento.choices, default=EstadoEvento.ACTIVO)

    imagen_portada = models.ImageField(upload_to='eventos/portadas/', null=True, blank=True)
    fecha_creacion = models.DateTimeField(auto_now_add=True)

    def __str__(self):
        return f"{self.titulo} ({self.get_estado_display()})"

    class Meta:
        db_table = 'evento'
        verbose_name = "Evento"
        verbose_name_plural = "Eventos"
        ordering = ['-fecha_inicio']

File: FrontendAlumni/src/models/Evento/Evento.ts
// src/models/Evento/Evento.ts

import type { Carrera } from "../Carrera/Carrera";
import type { Usuario } from "../Usuario/Usuario";

export type EventoEstado = "ACTIVO" | "FINALIZADO" | "CANCELADO";
export type EventoAlcance = "GLOBAL" | "CARRERAS";

export interface EventoParticipante extends Usuario {}

export interface Evento {
    id?: number;

    titulo?: string;
    descripcion?: string | null;

    fecha_inicio?: string | null;
    fecha_fin?: string | null;

    carreras?: number[];
    usuarios?: number[];

    requiere_registro?: boolean;
    estado?: EventoEstado;
    estado_display?: string | null;

    imagen_portada?: string | null;
    fecha_creacion?: string | null;

    // Campos derivados nuevos desde backend
    inscritos_count?: number;
    esta_inscrito?: boolean;
    alcance?: EventoAlcance;
    carreras_detalle?: Carrera[];
    puede_gestionar?: boolean;
    es_visible_para_mi?: boolean;
}